tools/testdata-scripts/globs.rs.data

```diff
@@ -28,8 +28,6 @@ const CLDR_JSON_GLOB: &[&str] = &[
     "cldr-dates-full/main/$LOCALES/timeZoneNames.json",
     "cldr-dates-full/main/cs/ca-gregorian.json",
     "cldr-dates-full/main/cs/timeZoneNames.json",
-    "cldr-dates-full/main/en-CA/ca-gregorian.json", // alt-variant in skeletons
```

Comment:
Observation: Someone at one point thought it was useful to test the alt-variant in skeletons, which I presume is covered during testdata datagen (as well as full datagen). Why delete this?

Reply:
Deleting this doesn't result in test failures or a testdata diff

Reply:
I think this is about test coverage for the date transformer on data with `-alt-variant` in skeletons, so deleting it isn't expected to result in test failures or a testdata diff. You're only reducing test coverage.

The justification for removing it is that we test full datagen in CI and we run it more often when generating compiled data, so we probably don't need this extra testdata as much as we did before.